src/TextParser.php

```diff
@@ -5,27 +5,37 @@
 use DirectoryIterator;
 use Monolog\Handler\StreamHandler;
 use Monolog\Logger;
+use Psr\Log\LoggerAwareTrait;
 
 class TextParser
 {
+    use LoggerAwareTrait;
     /** @var DirectoryIterator; Templates Directory Iterator */
     protected $directoryIterator;
 
-    /* @var Logger; Monolog Logger */
-    protected $logger;
+
 
 
     /**
      * @param string $templatesDir ; The path to the template files directory
      */
-    public function __construct($templatesDir)
+    public function __construct($templatesDir, $logger = null)
     {
-        $this->logger = new Logger('text-parser');
-        $this->logger->pushHandler(new StreamHandler('logs/text-parser.log', Logger::DEBUG));
+        if(!($logger instanceof Logger)) {
```

Comment:
shouldn't this be:
`if(!($logger instanceof LoggerInterface)) {`?

___________________________________

Anyway I would suggest you force it in the method deceleration: 

`public function __construct($templatesDir, LoggerInterface $logger = null)`

and just have a fallback to Monolog Logger in the constructor if the optional parameter is not provided:

```
if(empty($logger)) {
	$this->logger = new Logger('text-parser');
	$this->logger->pushHandler(new StreamHandler('logs/text-parser.log', Logger::DEBUG));
}
```

what do you think ?

Reply:
Ye, sure, my mistake. Of course it should be LoggerInterface. I considered both options, but i chose the option without forcing. I thought that it will be safer for older versions of PHP, but now I'm thinking that option with forcing is better.